bee-ternary/src/bigint/split_integer.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+//! Split integers to access the high and low parts of an integer.
+
+/// Represents a split integer into high and low parts.
+pub(crate) trait SplitInteger: Copy {
+    /// The type of the high part of the integer.
+    type High;
+    /// The type of the low part of the integer.
+    type Low;
+
+    /// Returns the high part of the integer.
+    fn hi(self) -> Self::High;
+    /// Returns the low part of the integer.
+    fn lo(self) -> Self::Low;
```

Comment:
I hate that `High` has one character more than`Low`. The trait designer obviously too as you can see in the method names. I propose to be consistent. You know what I'ld prefer 😉

Reply:
`hi`/`lo` are just commonly used and sort for that kind of purpose. I really don't want to change the type names so if you really mind I can change the names of the methods but tbh I don't think it makes much sense.

Reply:
no, I don't really really mind (that hard). Do what you think is fine 🙂